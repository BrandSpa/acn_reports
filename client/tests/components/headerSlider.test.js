import React from "react";
import { shallow, mount } from "enzyme";
import sinon from 'sinon';
import HeaderSlider from "../../components/headerSlider";

describe("header slider component", () => {
	it('should render as expected', () => {
		let wrapper = shallow(<HeaderSlider slides={[{id: 1}]} />);
		expect(wrapper.getNodes()).toMatchSnapshot();
	})

	it('should change slide to next', () => {
		let wrapper = shallow(<HeaderSlider slides={[{title: 'acn pope'}, {title: 'acn pope'}, {title: 'acn pope'}]} />);
		wrapper.instance().nextSlide();
		wrapper.instance().nextSlide();
		expect(wrapper.state()).toEqual({currentSlide: 2, left: '-200%'});
	})

	it('should change slide to prev', () => {
		let wrapper = shallow(<HeaderSlider slides={[{title: 'acn pope'}, {title: 'acn pope'}, {title: 'acn pope'}]} />);
		wrapper.instance().nextSlide();
		wrapper.instance().nextSlide();
		wrapper.instance().prevSlide();
		expect(wrapper.state()).toEqual({currentSlide: 1, left: '-100%'});
	})

	it('should show video modal', () => {
			let wrapper = mount(<HeaderSlider slides={[{title: 'acn pope', 'is_video': true}]} />);
			wrapper.find('.slider__slide__link-zone').simulate('click');
			expect( wrapper.find('.modal--show').length).toEqual(1);
	})

});
