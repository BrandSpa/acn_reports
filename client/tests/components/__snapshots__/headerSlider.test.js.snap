// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header slider component should render as expected 1`] = `
Array [
  <div
    className="slider"
    data-jsx={1296177145}
    style={
        Object {
            "height": "auto",
          }
    }
>
    <div
        className="slider__viewport"
        data-jsx={1296177145}
        style={
            Object {
                "height": "auto",
                "left": "0",
                "width": "100%",
              }
        }
    >
        <headerSlide
            height="auto"
            id={1}
            width="100%"
        />
    </div>
    <_class
        css=".slider[data-jsx=\\"1296177145\\"]{height:100%}.slider__viewport[data-jsx=\\"1296177145\\"]{width:100%;height:100vh;left:0}"
        styleId={1296177145}
    />
</div>,
]
`;
